src/mem/freelist.h

```diff
@@ -103,6 +90,22 @@ namespace snmalloc
       return p.template as_static<FreeObject>();
     }
 
+    /**
+     * Encode next
+     */
+    inline static CapPtr<FreeObject, CBAlloc> encode_next(
+      address_t curr, CapPtr<FreeObject, CBAlloc> next, const FreeListKey& key)
+    {
+      UNUSED(curr); // Used for alternative encoding schemes.
+#ifdef SNMALLOC_CHECK_CLIENT
+      // TODO don't do this on CHERI.
+      return CapPtr<FreeObject, CBAlloc>(address_cast(next) ^ key.key_next);
```

Comment:
I think that "don't do this on CHERI" can be pronounced as...

```suggestion
      if constexpr (aal_supports<StrictProvenance>)
      {
        UNUSED(key);
        return next;
      }
      return CapPtr<FreeObject, CBAlloc>(address_cast(next) ^ key.key_next);
```

Reply:
Thanks. I change that slightly, as it is typically best to use an `else` with `constexpr`.